### DISCUSSION TITLE
Question about the usage of save_dir in config files (1.x branch)

### DISCUSSION THREAD

**Discussion by i-aki-y** [open-mmlab/mmediting#1596] [General]
In the dev-branch (1.x), I noticed that some config files have a save_dir variable that is not used anywhere in the same config file.

ex:
    swinir/swinir_gan-x4s64w8d6e180_8xb4-lr1e-4-600k_df2k-ost.py 
    dim/dim_stage2-v16-pln_1xb1-1000k_comp1k.py
    ...

I think it expects to overwrite the save_dir in _base_/default_runtime.py, but the config mechanism does not support such a function that overwrites the parent config's local variable as far as I know.
Should not the config files, including unused save_dir definition, be fixed in the following ways?

```py
save_dir = "./work_dirs/"
...
default_hooks = dict(
    checkpoint=dict(
        type='CheckpointHook',
        out_dir=save_dir,
	...
    ),
    ...
```

**Comment by LeoXing1996** [comment]
Hey @i-aki-y, you are right. If you want to change `save_dir`, you should fix the config as you mentioned. 

We are planning to refactor the config system. Please keep an eye on [MMEngine](https://github.com/open-mmlab/mmengine).

**Comment by i-aki-y** [comment]
@LeoXing1996 Thank you for your clarification, I got it.
